opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/__init__.py

```diff
@@ -196,24 +213,52 @@ def create_observable_counter(
             self._instrument_id_instrument[instrument_id] = instrument
             return instrument
 
-    def create_histogram(self, name, unit="", description="") -> APIHistogram:
+    def create_histogram(
+        self,
+        name: str,
+        unit: str = "",
+        description: str = "",
+        advisory: Optional[MetricsHistogramAdvisory] = None,
+    ) -> APIHistogram:
+        if advisory is not None:
+            invalid_advisory = False
+            try:
+                boundaries = advisory.explicit_bucket_boundaries
+                invalid_advisory = not (
+                    boundaries
+                    and all(isinstance(e, float) for e in boundaries)
+                )
+            except (KeyError, TypeError):
+                invalid_advisory = True
+
+            if invalid_advisory:
+                advisory = None
+                _logger.warning(
+                    "Advisory must be a dict with explicit_bucket_boundaries key containing a sequence of numbers"
+                )
+
         (
-            is_instrument_registered,
             instrument_id,
-        ) = self._is_instrument_registered(name, _Histogram, unit, description)
+            status,
+        ) = self._register_instrument(
+            name, _Histogram, unit, description, advisory
+        )
 
-        if is_instrument_registered:
+        if status.conflict:
             # FIXME #2558 go through all views here and check if this
             # instrument registration conflict can be fixed. If it can be, do
             # not log the following warning.
             _logger.warning(
                 "An instrument with name %s, type %s, unit %s and "
-                "description %s has been created already.",
+                "description %s has been created already with a "
+                "different advisory value %s.",
```

Comment:
maybe the message should mention the consequence of this–that we will use the first advisory value instead.

Reply:
Fixed in 027da734b8354dd199da77dbd7aa3cf745377e76